base/src/main/java/io/spine/util/Preconditions2.java

```diff
@@ -37,31 +44,230 @@ private Preconditions2() {
         return stringToCheck;
     }
+
+    /**
+     * Ensures that the passed value is positive.
+     *
+     * @param value the value to check
+     * @throws IllegalArgumentException if the value is negative or zero
+     */
+    public static long checkPositive(long value) {
+        if (value <= 0) {
+            throw newIllegalArgumentException("A positive value expected. Passed: %d.", value);
+        }
+        return value;
+    }
+
+    /**
+     * Ensures that the passed value is positive.
+     *
+     * @param value
+     *         the value to check
+     * @param errorMessage
+     *         the exception message to use if the check fails
+     * @throws IllegalArgumentException
+     *         if the value is negative or zero
+     */
+    public static long checkPositive(long value, @Nullable String errorMessage) {
+        checkNotNull(errorMessage);
+        checkArgument(value > 0L, errorMessage);
+        return value;
+    }
+
+    /**
+     * Ensures that the passed message is not in the default state.
+     *
+     * @param message
+     *         the message to check
+     * @param <T>
+     *         the type of the message
+     * @return the passed message
+     * @throws IllegalArgumentException
+     *          if the passed message has the default state
+     * @throws NullPointerException
+     *          if the passed message is {@code null}
+     */
+    @CanIgnoreReturnValue
+    public static <T extends @NonNull Message> T checkNotDefaultArg(T message) {
+        checkArgument(!Messages.isDefault(message));
+        return message;
+    }
+
+    /**
+     * Ensures that the passed message is not in the default state.
+     *
+     * @param message
+     *         the message to check
+     * @param <T>
+     *         the type of the message
+     * @return the passed message
+     * @throws IllegalStateException
+     *          if the passed message has the default state
+     * @throws NullPointerException
+     *          if the passed message is {@code null}
+     */
+    @CanIgnoreReturnValue
+    public static <T extends @NonNull Message> T checkNotDefaultState(T message) {
+        checkState(!Messages.isDefault(message));
+        return message;
+    }
+
+    /**
+     * Ensures that the passed object is not in its default state and is not {@code null}.
+     *
+     * @param object
+     *         the {@code Message} instance to check
+     * @param errorMessage
+     *         the message for the exception to be thrown;
+     *         will be converted to a string using {@link String#valueOf(Object)}
+     * @throws IllegalStateException
+     *         if the object is in its default state
+     * @throws NullPointerException
+     *         if the passed message is {@code null}
+     */
+    @CanIgnoreReturnValue
+    public static <M extends Message>
+    M checkNotDefaultArg(M object, @Nullable Object errorMessage) {
+        checkNotNull(object);
```

Comment:
Shouldn't we pass our custom error message here (and below) as well?

Reply:
So what about the `errorMessage`? Why don't we pass it to the `notNull` check?

Reply:
Because it's nullable.

Reply:
So what? We still pass it to the `checkArgument` even though it still might be `null`. It's just inconsistent. Why do we omit the custom error message for `notNull` checks, but pass it to the `checkArgument`? We do the same in almost every method in this class.

Let's define some consistent behavior and use it across all methods. If we think that we should not pass the custom error message to the `checkNotNull` check, then maybe we should explicitly state it.